analyses/cell-type-ewings/exploratory_analysis/06-tumor-cell-state-assignment.Rmd

```diff
@@ -0,0 +1,606 @@
+      # cluster 8 has high MSC marker gene expression and a mix of MSC like cell types
+      library_id == "SCPCL000822" & cluster == 7 ~ "mesenchymal-like cell",
+      library_id == "SCPCL000824" & cluster %in% c(2, 3, 4, 5, 6, 7) ~ "tumor", 
+      library_id == "SCPCL000824" & cluster == 8 ~ "endothelial cell",
+      # cluster 9 has high immune markers and is mostly macrohpages
+      library_id == "SCPCL000824" & cluster == 9 ~ "macrophage",
+      # cluster 1 has high MSC marker gene expression and a mix of MSC like cell types
+      library_id == "SCPCL000824" & cluster == 1 ~ "mesenchymal-like cell",
+      # if they didn't have high expression of any markers, we label them as unknown
+      .default = "unknown"
+    )
+  )
+```
+
+```{r}
+library_ids |>
+  purrr::map(\(id){
+    cluster_classification_df |>
+      dplyr::filter(library_id == id) |>
+      ggplot(aes(x = UMAP1, y = UMAP2, color = cluster_classification)) +
+      geom_point(alpha = 0.5, size = 0.1) +
+      theme(
+        aspect.ratio = 1
+      ) +
+      labs(title = id) +
+      scale_color_brewer(palette = "Dark2") +
+      guides(color = guide_legend(override.aes = list(alpha = 1, size = 1.5)))
+  })
+```
+
+Looking at this I think the cells that are classified as tumor cells using the cluster assignments mostly line up with the `SingleR` annotations. 
+So we can label any cells from clusters that have high tumor marker gene expression as tumor cells. 
+
+### Look at marker gene expression for tumor cell states in all cells
+
+Below we look at the expression of genes for the tumor cell states across all cells. 
+This will show us how the expression of these gene sets varies across cells in each library and help ensure we have the correct population of tumor cells.
+
+The value for each of the marker gene sets corresponds to the sum of the expression of all genes in the given gene list. 
+(Note that I also looked at mean and it had similar output)
+
+The gene sets that we use are: 
+
+- `EWS-low`: Marker genes from `tumor-cell-state-markers.tsv` that correspond to the `EWS-low` cell state. 
+- `EWS-high`: Marker genes from `tumor-cell-state-markers.tsv` that correspond to the `EWS-high` cell state. 
+- `proliferative` : Marker genes from `tumor-cell-state-markers.tsv` that correspond to the `proliferative` cell state. 
+Note that a lot of the Ewing sarcoma literature argues that `EWS-FLI1` high cells are proliferative. 
+- `aynaud-ews-high`: Marker genes from `gene_signatures/aynaud-ews-targets.tsv` that correspond to genes activated by `EWS-FLI1`. 
+- `wrenn-ews-low`: Marker genes from `gene_signatures/wrenn-nt5e-genes.tsv` that correspond to genes repressed by `EWS-FLI1`. 
+
+```{r, fig.height=7, fig.width=10}
+# pull out columns that have sum of marker gene expression 
+marker_gene_cols <- colnames(cluster_classification_df)[stringr::str_detect(colnames(cluster_classification_df), "_sum$")]
+
+# look at the expression of all gene sets across both libraries 
```

Comment:
Looking at these plots, I'm curious if the cells with the high proliferative levels are actually a subset of the EWS-FL1 high cells or whether they are distinct cells within the same cluster. I think there are more high expression proliferative cells in the data since the plot suggests at first glance and lighter points are less frequent so they are hiding behind the darker points. If these are indeed mostly a subset, this could actually be a "sub-state" worth classifying. Might be worth a quick check, maybe a scatterplot or so?